specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,313 @@
+
+- IDropTarget::DragEnter
+- IDropTarget::DragMove
+- IDropTarget::DragLeave
+- IDropTarget::Drop
+
+For other UI frameworks such as Xaml, the hosting application would set event handlers for DragEnter, DragOver, DragLeave, and Drop
+on the UIElement that contains the WebView2 element.
+
+# Examples
+## Win32
+```c++
+// Win32 Sample
+
+// Implementation for IDropTarget
+HRESULT DropTarget::DragEnter(IDataObject* dataObject,
+    DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we entered so it can update the drag image and get
+    // the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->DragEnter(GetHWND(), dataObject, &point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->DragEnter(
+        dataObject, keyState, point, effect);
+}
+
+HRESULT DropTarget::DragOver(DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we moved over it so it can update the drag image
+    // and get the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->DragOver(&point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    // This returns whether the resultant point is over the WebView visual.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->DragOver(
+        keyState, point, effect);
+}
+
+HRESULT DropTarget::DragLeave()
+{
+    // Tell the helper that we moved out of it so it can update the drag image.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->DragLeave();
+    }
+
+    return m_webViewCompositionController2->DragLeave();
+}
+
+HRESULT DropTarget::Drop(IDataObject* dataObject,
+    DWORD keyState,
+    POINTL point,
+    DWORD* effect)
+{
+    POINT point = { cursorPosition.x, cursorPosition.y };
+    // Tell the helper that we dropped onto it so it can update the drag image
+    // and get the correct effect.
+    wil::com_ptr<IDropTargetHelper> dropHelper = DropHelper();
+    if (dropHelper)
+    {
+        dropHelper->Drop(dataObject, &point, *effect);
+    }
+
+    // Convert the screen point to client coordinates add the WebView's offset.
+    // This returns whether the resultant point is over the WebView visual.
+    m_viewComponent->OffsetPointToWebView(&point);
+    return m_webViewCompositionController2->Drop(
+        dataObject, keyState, point, effect);
+}
+```
+
+## WinRT
+```c#
+// WinRT Sample
+private void WebView_DragEnter(object sender, DragEventArgs e)
```

Comment:
Where are these `DragEventArgs` from (what namespace)?

Reply:
Switching to WinRT as suggested in another conversation. We don't need these.